nhaystack-rt/src/nhaystack/server/tags/BNSiteRefRelation.java

```diff
@@ -56,24 +58,32 @@
      * BControlPoint.
      */
     @Override
-    public Optional<Relation> getRelation(Entity point)
+    public Optional<Relation> getRelation(Entity source)
     {
-        if (!(point instanceof BControlPoint))
+        if (!(source instanceof BControlPoint))
         {
             return Optional.empty();
         }
 
         Id relationId = getRelationId();
-        if (!hasDirectRelation((BComponent) point, relationId))
+        if (hasDirectRelation((BComponent) source, relationId))
         {
-            BComponent equip = getEquip((BControlPoint) point);
-            if (equip != null)
+            return Optional.empty();
+        }
+
+        BComponent equip = getEquip((BControlPoint) source);
+        if (equip == null)
+        {
+            return Optional.empty();
+        }
+
+        Optional<Relation> equipSiteRef = equip.relations().get(relationId, Relations.OUT);
+        if (equipSiteRef.isPresent())
+        {
+            Entity equipSite = equipSiteRef.get().getEndpoint();
+            if (hasSiteTag(equipSite))

```

Comment:
This is a small defect fix- the endpoint of the siteRef relation must have a site tag.  We would have implied an outbound relation to the site component but not any inbound relations from points.